src/Draco.Compiler/Internal/Binding/Binder_BoundLvalue.cs

```diff
@@ -29,4 +33,57 @@ internal partial class Binder
 
     private BoundLvalue TypeGlobalLvalue(UntypedGlobalLvalue global, ConstraintSolver constraints, DiagnosticBag diagnostics) =>
         new BoundGlobalLvalue(global.Syntax, global.Global);
+
+    private BoundLvalue TypeFieldLvalue(UntypedFieldLvalue field, ConstraintSolver constraints, DiagnosticBag diagnostics)
+    {
+        if (!field.Field.IsMutable)
+        {
+            diagnostics.Add(Diagnostic.Create(
+                template: SymbolResolutionErrors.CannotAssignToReadonlyOrConstantField,
+                location: field.Syntax?.Location,
+                field.Field.FullName));
+            return new BoundIllegalLvalue(field.Syntax);
+        }
```

Comment:
We should check this in flow analysis instead, that's where we check local/global assignment too